docs/book/src/developer/providers/v1.1-to-v1.2.md

```diff
@@ -31,4 +31,7 @@ in ClusterAPI are kept in sync with the versions used by `sigs.k8s.io/controller
 
 ### Other
 
--
+- KubeadmControlPlane controller and DockerMachineMachine controller no longer block if `cluster.status.infrastructureReady` is false.
+This fixes the following problems:
+  - Cluster deletion leaves behind orphaned machine templates referenced in KCP.
+  - CAPD cluster deletion is blocked if infrastructure is not ready.  
```

Comment:
I think that we should use this section to document only what the providers are expected to do or things that can impact them while upgrading (it is not required to document all the changes applied to KCP/CAPD if they do not impact providers)

what about rephrasing in.

```suggestion
after investigating the issue... we discovered that improper implementation of a check on cluster.status.infrastructureReady can lead to problems during cluster deletion.

as a consequence, we recommend all the providers check that
- the check for cluster.status.infrastructureReady=true usually existing at the beginning of the reconcile loop for control-plane providers is implemented after setting external objects ref and 
- the check for cluster.status.infrastructureReady=true usually existing  at the beginning of the reconcile loop for infrastructure provider does not prevent the object to be deleted
```